typing/ikinds/ldd_hybrid_opt.ml

```diff
@@ -0,0 +1,665 @@
+(* Lattice-valued ZDDs.
+
+   Big picture: we represent LDDs as a ZDD-like DAG ordered by [var] id.
+   A [node_block] is the lattice expression lo ⊔ (var ⊓ hi), with the
+   invariant that all vars appearing in [lo] and [hi] are strictly greater
+   than the node's [var]. We also maintain canonical form [hi = hi - lo],
+   so [hi] is disjoint from [lo]. *)
```

Comment:
Thanks, this is a very helpful comment. A couple notes:
- Can you say what LDD and ZDD stand for? (And if you have one, include a link to somewhere someone can read more about them)
- I think everything after "[node_block]" better belongs at the definition of the `node_block` type, since that's the place people will likely look for this sort of information.
- "[lo] and [hi] are strictly greater than the node's [var]" - it's not clear here that by "greater" you mean "the id of the var is greater" and not "the value of the var is greater".